egs/librispeech/asr/simple_v1/espnet_utils/numericalizer.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Xiaomi Corporation (Author: Guo Liyong)
+# Apache 2.0
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Iterable, List, Optional, Union
+from pathlib import Path
+
+import numpy as np
+
+from torchtext.data.functional import load_sp_model
+
+
+class NumericalizerMixin(ABC):
+
+    def _assign_special_symbols(self):
+        # <sos> and <eos> share same index for model download from espnet model zoo
+        assert '<sos/eos>' in self.token2idx \
+                or ('<sos>' in self.token2idx and '<eos>' in self.tokenid)
+        assert '<unk>' in self.token2idx
+        self.sos_idx = self.token2idx[
+            '<sos/eos>'] if '<sos/eos>' in self.token2idx else self.token2idx[
+                '<sos>']
+        self.eos_idx = self.token2idx[
+            '<sos/eos>'] if '<sos/eos>' in self.token2idx else self.token2idx[
+                '<eos>']
+        self.unk_idx = self.token2idx['<unk>']
+
+
+@dataclass
```

Comment:
Do we really need to use `dataclass` here?

Also, could you remove the class `NumericalizerMixin`?
The extra level of inheritance makes the code hard to read.